python/orca/src/bigdl/orca/learn/tf2/tf_runner.py

```diff
@@ -196,6 +196,13 @@ def dataset_fn(input_context):
         return dataset
 
     def _handle_sharding(self, dataset):
+        import tensorflow as tf
+        if isinstance(dataset, tf.data.Dataset):
+            options = tf.data.Options()
+            # Disable Tensorflow autosharding since the dataset has already been sharded.
+            options.experimental_distribute.auto_shard_policy = \
+                tf.data.experimental.AutoShardPolicy.OFF
+            dataset = dataset.with_options(options)
```

Comment:
What if users customize a data creator function?  In that case, the dataset might not been sharded.
Shall we just put this part to the data creator function for ray dataset?

Reply:
Sure. That's a good idea.😄

Reply:
> What if users customize a data creator function? In that case, the dataset might not been sharded. Shall we just put this part to the data creator function for ray dataset?

Done.